projects/js-packages/svelte-data-sync-client/src/utils.ts

```diff
@@ -0,0 +1,38 @@
+import { z } from 'zod';
+
+/**
+ * JSON Schema form Zod
+ * Straight out of the docs:
+ * https://github.com/colinhacks/zod
+ */
+
+const literalSchema = z.union( [ z.string(), z.number(), z.boolean(), z.null() ] );
+type Literal = z.infer< typeof literalSchema >;
+type Json = Literal | { [ key: string ]: Json } | Json[];
+
+export const jsonSchema: z.ZodType< Json > = z.lazy( () =>
+	z.union( [ literalSchema, z.array( jsonSchema ), z.record( jsonSchema ) ] )
+);
+export type JSONSchema = z.infer< typeof jsonSchema >;
+
+/**
+ * Tries to stringify an object, returns the original value if it cannot.
+ *
+ * @param {JSONSchema | string} value - The value to stringify.
+ * @returns {*} - The stringified value or the original value.
+ */
+export function maybeStringify< T >( value: JSONSchema | string ): string | T {
```

Comment:
What's the purpose of `<T>`? It sounds like it will let you coerce `value` to be any type you want?

Reply:
Ah, this is some legacy code that crept in. This is only necessary for `request` and that's been updated to send everything as JSON anyway, so the values are always stringified now.